x-pack/plugins/snapshot_restore/public/app/sections/home/home.tsx

```diff
@@ -99,8 +99,16 @@ export const SnapshotRestoreHome: React.FunctionComponent<Props> = ({
         <EuiSpacer size="m" />
 
         <Switch>
-          <Route exact path={`${BASE_PATH}/repositories/:name*`} component={RepositoryList} />
-          <Route exact path={`${BASE_PATH}/snapshots/:name*`} component={SnapshotList} />
+          <Route
+            exact
+            path={`${BASE_PATH}/repositories/:repositoryName*`}
+            component={RepositoryList}
+          />
+          <Route
+            exact
+            path={`${BASE_PATH}/snapshots/:repositoryName?/:snapshotId*`}
```

Comment:
this routing doesn't work for repository names with `/` in them. for example, set up a repository with the name `foo/bar` and add a snapshot into it. the detail panel will encounter a 404 error

splitting this route into two routes, one with a greedy capture on `repositoryName` seems to work:
```jsx
          <Route
            exact
            path={`${BASE_PATH}/snapshots`}
            component={SnapshotList}
          />
          <Route
            exact
            path={`${BASE_PATH}/snapshots/:repositoryName*/:snapshotId`}
            component={SnapshotList}
          />
```

snapshot IDs have ES character restrictions, so we don't need to worry about `/` for them 😄